tensorflow/compiler/xla/service/mlir_gpu/lhlo_dialect_emitter.cc

```diff
@@ -254,6 +255,20 @@ Status LhloDialectEmitter::HandleBroadcast(HloInstruction* instr) {
   return Status::OK();
 }
 
+Status LhloDialectEmitter::HandleConcatenate(HloInstruction* instr) {
+  mlir::IntegerAttr concatenate_dim = builder_.getI64IntegerAttr(
+      static_cast<HloConcatenateInstruction*>(instr)->concatenate_dimension());
+
+  TF_ASSIGN_OR_RETURN(auto function, CreateFunction(*instr));
+  OpBuilder func_builder(function.getBody());
+  llvm::SmallVector<Value, 4> arg_values{function.args_begin(),
+                                         function.args_end()};
+  Value result_memref = function.getArgument(function.getNumArguments() - 1);
+  func_builder.create<lhlo::ConcatenateOp>(
+      getLocation(instr), arg_values, result_memref, concatenate_dim);
```

Comment:
This seems to pass the result memref twice. How about `function.getArguments().drop_back(1)` for the arguments and `function.getArguments().back()`? These (or similar) should exist.

Reply:
Thanks, I made a mistake here.